.github/workflows/tests.yml

```
@@ -0,0 +1,56 @@
+        xvfb-run --auto-servernum npm run test
+        xvfb-run --auto-servernum npm run test:unit
+
+    - name: Run tests (macOS/Windows)
+      if: runner.os != 'Linux'
+      run: |
+        npm run test
+        npm run test:unit
+
+    - name: Upload test results
+      if: always()
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)